ui/components/app/assets/nfts/nfts-items/nfts-items.js

```diff
@@ -116,39 +118,39 @@ export default function NftsItems({
     dispatch,
   ]);
 
-  const history = useHistory();
+  const getAssetImageUrlAndUpdate = async (image, nft) => {
+    const nftImage = await getAssetImageURL(image, ipfsGateway);
+    const updatedNFt = {
+      ...nft,
+      ipfsImageUpdated: nftImage,
+    };
+    return updatedNFt;
+  };
 
-  const renderCollectionImage = (collectionImage, collectionName) => {
-    if (collectionImage?.startsWith('ipfs') && !ipfsGateway) {
-      return (
-        <div className="nfts-items__collection-image-alt">
-          {collectionName?.[0]?.toUpperCase() ?? null}
-        </div>
-      );
-    }
-    if (!openSeaEnabled && !collectionImage?.startsWith('ipfs')) {
-      return (
-        <div className="nfts-items__collection-image-alt">
-          {collectionName?.[0]?.toUpperCase() ?? null}
-        </div>
-      );
-    }
+  useEffect(() => {
+    const promisesArr = [];
+    const modifyItems = async () => {
+      for (const key of collectionsKeys) {
+        const { nfts } = collections[key];
+        for (const singleNft of nfts) {
+          const { image, imageOriginal } = singleNft;
 
-    if (collectionImage) {
-      return (
-        <img
-          alt={collectionName}
-          src={getAssetImageURL(collectionImage, ipfsGateway)}
-          className="nfts-items__collection-image"
-        />
-      );
-    }
-    return (
-      <div className="nfts-items__collection-image-alt">
-        {collectionName?.[0]?.toUpperCase() ?? null}
-      </div>
-    );
-  };
+          const isImageHosted = image?.startsWith('https:');
```

Comment:
cleartext `http:` might still be a thing?
```suggestion
          const isImageHosted = image?.startsWith('https:') || image?.startsWith('http:');
```

should `file:` be considered...?

Reply:
Would an image hosted on http be less secure than a one on https? In which case i assume maybe we dnt want to show it; 
Good question for `file:`, i really haven't yet seen such case, these images come from our third party provider and they are hosted on their domain that starts with `https`, in 99% of the time we will display the returned image, but there are some rare case where we fallback to the originalImage, which is the image fetched from the token URI, and usually they are hosted on IPFS;

Reply:
I'd think it'd make sense to treat `https:`, `http:`, and `file:` the same, here. Sure http is a less secure transport than https, but also probably not the concern of this module..?

Reply:
Right, i agree that if there is a concern it should not be the concern of this component;
I will commit your suggested change : 
` const isImageHosted = image?.startsWith('https:') || image?.startsWith('http:');`

I am a bit hesitant to commit the `file:` change because i would like to see a use case and test it before making the change, would that be okay with you?

I have checked in the codebase if there is anything that checks if a certain url is "secure" or not, i do not think there is; we do check if a url is [valid](https://github.com/MetaMask/core/blob/fe3dfcb5f71803c103ab7e6d91f91bcac2ae9c3c/packages/network-controller/src/NetworkController.ts#L693) or not in core or [extension](https://github.com/MetaMask/metamask-extension/blob/9f8651c95b6b4aabbce88ae375b15b0024d20d50/app/scripts/lib/util.ts#L254) when necessary;